### DISCUSSION TITLE
Installing mxnet on PowerPC

### DISCUSSION THREAD

**Discussion by Talgin** [apache/incubator-mxnet#15553] [Q&A] [answered]
Note: Providing complete information in the most concise form is the best way to get help. This issue template serves as the checklist for essential information to most of the technical issues and bug reports. For non-technical issues and feature requests, feel free to present the information in what you believe is the best form.

For Q & A and discussion, please start a discussion thread at https://discuss.mxnet.io 

## Description
(Brief description of the problem in no more than 2 sentences.)
We are having troubles with installing mxnet on powerPC
## Environment info (Required)
Platform: Power9
OS: Redhat Linux
Cuda: 10.1
Python: python2.7

We tried to install the mxnet-cu101 package for python2.7 on our powerpc machine, but pip cannot find the version: 
"ERROR: Could not find a version that satisfies the requirement mxnet-cu101 (from versions: none)
ERROR: No matching distribution found for mxnet-cu101"

1. What should we do? 
2. We wanted to run it with python, should we have to build it from source and use python bindings?
3. Or do you have version for powerpc? :) 

Thank you!

[a comment by an automated account is not shown]

**Comment by marcoabreu** [comment]
Sorry, but we are not officially supporting PowerPC. Please try compiling yourself.

**Comment by keerthanvasist** [comment]
@mxnet-label-bot add [question]

**Comment by pasa13142** [comment]
@marcoabreu Hello, is there any guide we can follow for building mxnet on power9 ?

**Comment by leezu** [comment, marked as answer]
You may follow https://mxnet.apache.org/get_started/ubuntu_setup  in analogy

**Comment by Talgin** [comment]
Yeah, this guide helped us too. 

> You may follow https://mxnet.apache.org/get_started/ubuntu_setup in analogy

**Comment by leezu** [comment]
If the guide needs any extension to cover PowerPC, feel free to open a PR. Thanks

**Comment by pasa13142** [comment]
@leezu @Talgin , Thanks for helping guys, It worked !

But there is little problem i got, my progress takes 1 sec on 1080ti on 64 bit intel, 1.2 sec on v100 on ppc64.

v100 on ppc should be faster than 1080ti on 64 bit. You guys have any idea ?

**Comment by pasa13142** [comment]
@leezu @Talgin no idea how can i fix this ? :(

**Comment by leezu** [comment]
You need to investigate and find some minimal example that shows the problem. For example, by using the benchmark scripts https://github.com/apache/incubator-mxnet/tree/master/benchmark/opperf

**Comment by Talgin** [comment]
@pasa13142, we had the same problem, training was slower on power9 with Tesla V100  than on my workstation. I do not remember if they found a reason, but our server management service said that ppc was not originally made to do such things (training) :) Like they are good at calculation, maybe some kind of server operations, but they (ibms ppc) do not have much experience with GPUs. 
I would also be happy if you share the solution, if you found one.

**Comment by pasa13142** [comment]
Thank you, if i find out, i'll update, but unfortunately signs are not good on ibms ppc :(

**Comment by pasa13142** [comment]
@Talgin hey! Which math library did you select for powerpc ? And does it work or does it matter for gpu ? what could be make v100 slower than 1080 ? :/

**Comment by Talgin** [comment]
Hi, @pasa13142! We used numpy, scikit-learn, pandas and many other. Installation was without problems and they worked OK. Have you found something interesting? :)

**Comment by Talgin** [comment]
Hi, @pasa13142 
Try to directly connect your monitor to the server (not with ssh), sometimes the network can cause fluctuations.
